### PR TITLE
feat: add image dark/light mode and text

### PR DESCRIPTION
**Pull request by tisidro**
FSA22V2-247-styling-listing-page-theming1

FSA22V2-247

### Description:

-add light/dark mode image support
-fix card text so it is black on home page

### Spec:

Designs: [Designs](DESIGN_URL)

See Story: [FSA22V2-ISSUE](https://sparkbox.atlassian.net/browse/FSA22V2-ISSUE)

### Validation:

- [ ] This PR has code changes, and our linters still pass.
- [ ] This PR affects production code, so it was browser tested (see below).
- [ ] This PR has new code, so new tests were added or updated, and they pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. Run `npm install` to make sure you have all proper dependencies. (this project requires Node 16+)
4. Copy the `.env.example` file, making sure it's in the root of your project, and rename it `.env.local`. Search for "Accessible Components" in 1Password to find the API Key and Base ID. Add both of those secrets to your newly created `.env.local` file. That should establish your connection to our Airtable database.
5. Run `npm run lint` to confirm no errors.
6. Run `npm run test` to confirm all tests pass.
7. In terminal: `npm run dev`.
8. Check [localhost:3000](http://localhost:3000/) to see changes.
9. Run axe Devtools on affected pages to confirm no urgent errors.

### Pending Questions:

1.
2.

---

#### Browser Testing

In these browsers, behavior & design closely match original specifications. A user is able to access all content and functionality, including the usability of required assistive devices, such as keyboard and screenreader.

**macOS**

- [ ] Safari (last 2 major versions)
- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)

**Windows**

- [ ] Chrome (last 6 months)
- [ ] Firefox (last 6 months)
- [ ] Edge (last 6 months)

**Mobile**

- [ ] Safari (last 2 major versions)
- [ ] Chrome on Android (last 6 months)
- [ ] Firefox on Android (last 6 months)

### CODE REVIEW

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]
Hey Terri- nice work with this! I noticed some issues with the footer links, but the changes I was going to suggest, are actually taken care of in the listing page style PR I'm almost done with! 🥳

I did notice that we're missing the borders of the component card headings. I don't think this was a change made in this branch, but I am pointing this one out because it's got a `var` that should change. 

At about line 36: 
```
&__header {
// this change should make the borders black
  border-top: 0.25rem solid var(--background-color);
}
```

I think the other missing styles (like the padding and the extra grid item borders) get fixed on my listing page styles branch. Nice work on this!

**Comment by marissahuysentruyt** [comment]
Also just noticed- is there a way you can invert the colors for dark mode for the Sparkbox logo? It's black, and with the black background in the hero section, we can't see it! We might even need to get a white version and so a similar `source` like you did for the other image.

File: src/styles/components/_footer.scss [hunk 1]
```diff
@@ -1,42 +1,58 @@
...
   }
 
-  &__item a {
+  &__item > a {
     text-decoration: underline;
-    color: map.get($colors, "black");
-    background: linear-gradient(to left, rgb(0 0 0 / 0%) 50%, map.get($colors, accent-teal) 50%);
+    color: var(--font-color);
+    background: linear-gradient(to left, transparent, var(--accent-color) 50%);
     background-size: 201% 100%;
```
Comment (marissahuysentruyt) [inline src/styles/components/_footer.scss:52]: line 51 should be: 
`background: linear-gradient(to left, transparent 50%, var(--accent-color) 50%);`

This background needs an additional 50% so that we don't see the pink on the footer links BEFORE we actually do a hover.

File: src/styles/components/_header.scss [hunk 2]
```diff
@@ -1,7 +1,37 @@
...
+      width: 100%;
+      height: auto;
+      border: 2px solid transparent;
+      transition: all 150ms ease-in-out;
+    }
+
+    &:focus img {
+      border: 2px solid var(--accent-color);
+    }
+  }
```
Comment (marissahuysentruyt) [inline src/styles/components/_header.scss:24]: Looks like here you have the `--accent-color`, but the focus for the Sparkbox logo is still the teal. Did you want it to be pink?

File: src/styles/components/_header.scss [hunk 3]
```diff
@@ -1,7 +1,37 @@
...
+  @media only screen and (min-width: $breakpoint-tablet) {
+    padding: 1rem 1.25rem 0 8.5625rem; // top: 16px/16; right: 20px/16; bottom: 0; left: 137px/16
+  }
+
+  @media only screen and (min-width: 100rem) {
+    margin: auto;
+  }
+
+  @media screen and (prefers-color-scheme: dark) {
+    filter: invert(100);
```
Comment (marissahuysentruyt) [inline src/styles/components/_header.scss:35]: Beautiful!! So glad the invert worked for the header and footer!!!

File: src/styles/components/_header.scss (hunk 2 above)
Comment (tisidro) [reply src/styles/components/_header.scss:24]: Not sure why that would be as the root file specifies --accent-color to change to pink for dark mode. I'll go over this w/Noelle to make sure but it should be working.

image

File: src/styles/components/_footer.scss (hunk 1 above)
Comment (tisidro) [reply src/styles/components/_footer.scss:52]: Thanks, I was wondering why that looked funny! I fixed that :)

File: src/styles/components/_related-components.scss [hunk 4]
```diff
@@ -1,10 +1,11 @@
 $breakpoint-tablet: 900px;
 
 .rel-component {
```
[the remark this answers is not in the document]
Comment (tisidro) [reply src/styles/components/_related-components.scss:3]: I think we should discuss in a slack message. I'll message the apprentices and ask. 

I am not sure if this is in scope for this PR though as it would have to be changed on several pages. Is this something you or Jared can fix on your "style-tweaking" PR? 

@dustin-jw what do you think? Thanks! :)

[the remark this answers is not in the document]
Comment (dustin-jw) [reply src/styles/components/_related-components.scss:3]: I'll add "renaming breakpoint variables" to my list of cards to write 😄

**Review by marissahuysentruyt — COMMENTED** [review]
Is there any way to work that image to be more responsive? it's not shrinking with the container. You can see in the screenshot that it's still it's entire size, while the rest of the screen has shrunk down off to the left: 
Screen Shot 2022-11-04 at 10 25 49 AM

File: src/styles/components/_component-card.scss [hunk 5]
```diff
@@ -6,13 +6,14 @@
 
   .component-card {
     margin: 0;
-    border: 2px solid black;
+    border: 2px solid var(--foreground-color);
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:9]: This is that border I think I mentioned to you. Right now, the border is white, but it should stay black (at least according to the designs) 

`border: 2px solid var(--background-color);`

File: src/styles/components/_component-card.scss [hunk 6]
```diff
@@ -21,23 +22,55 @@
...
+      font-size: 1.125rem; // 18px/16
       line-height: 4.6667rem; // 75px
-      letter-spacing: 2%;
-      text-align: center;
-      display: block;
-      border-top: 0.25rem solid map.get($colors, "black");
+      letter-spacing: 0.02em;
+      border-top: 0.25rem solid var(--foreground-color);
+      background-color: var(--background-color);
+      position: relative;
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:50]: Same thing here: Change the `border-top` to black/background-color; change the 'background-color` to white/foreground-color.

File: src/styles/components/_header.scss (hunk 2 above)
Comment (marissahuysentruyt) [reply src/styles/components/_header.scss:24]: Yeah you'll have to check with her! 
Screen Shot 2022-11-04 at 10 41 22 AM

File: src/styles/components/_component-grid.scss [hunk 7]
```diff
@@ -69,6 +179,10 @@ $breakpoint-tablet: 820px;
       &--max-scott {
         grid-area: max-scott;
       }
+
+      border: 1px solid;
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-grid.scss:183]: Oh I found those stray borders!! Right now, we get like, a secondary white border on all the component cards, so I would just delete this.

File: src/styles/components/_component-grid.scss [hunk 8]
```diff
@@ -69,6 +179,10 @@ $breakpoint-tablet: 820px;
...
+      text-align: center;
+      align-items: center;
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-grid.scss:185]: These are actually taken care of elsewhere, so you could delete these too I think.

File: src/styles/components/_component-card.scss (hunk 6 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:50]: I just switched these! :)

File: src/styles/components/_component-card.scss (hunk 5 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:9]: Got it!

File: src/styles/components/_component-grid.scss (hunk 7 above)
Comment (tisidro) [reply src/styles/components/_component-grid.scss:183]: Nice catch, just got rid of it! :)

File: src/styles/components/_component-grid.scss (hunk 8 above)
Comment (tisidro) [reply src/styles/components/_component-grid.scss:185]: Yep, they were not needed so they are deleted now!

File: src/styles/components/_related-components.scss (hunk 4 above)
[the remark this answers is not in the document]
Comment (tisidro) [reply src/styles/components/_related-components.scss:3]: Great, thank you! I'll leave the breakpoints as is for this card then. :)

File: src/pages/index.page.jsx [hunk 9]
```diff
@@ -17,13 +17,16 @@ export default function Home() {
                 we&#39;re adding new components frequently!
               </p>
             </div>
-            <div>
+            <picture className="home-top__top-image">
+              <source
+                srcset="/images/home-header-image_dark.svg"
+                media="(prefers-color-scheme: dark)"
+              />
```
Comment (marissahuysentruyt) [inline src/pages/index.page.jsx:24]: If I remove that `className` from `picture`, and put that `className` on the image instead, it behaves just like light-mode version. 🤷‍♀️

I know you said there had to be the class on `picture`, so I'm not sure why that works. This might just be a quick fix before we do a card for the images like we talked about during our rebasing.

File: src/styles/components/_component-card.scss [hunk 10]
```diff
@@ -6,14 +6,14 @@
...
-    border: 2px solid var(--foreground-color);
+    border: 2px solid var(--background-color);
     display: flex;
     flex-direction: column-reverse;
     height: 21.875rem; // 350px/16
     width: 100%;
     cursor: pointer;
     transition: all 150ms ease-in-out;
-    color: var(--font-color-dark);
+
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:16]: You can probably delete the extra line here (not that it's really hurting anything)

File: src/styles/components/_component-card.scss [hunk 11]
```diff
@@ -6,14 +6,14 @@
...
 
     @media (prefers-color-scheme: dark) {
       border: 2px solid var(--background-color);
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:19]: Oh actually, I think I found an issue here: If I switch back and forth between light and dark mode, some of the styles are missing from one or the other. Here's what I did to fix it. 

This starts at about line 9 (to change the border color first): 
```
    border: 2px solid var(--background-color);
    display: flex;
    flex-direction: column-reverse;
    height: 21.875rem; // 350px/16
    width: 100%;
    cursor: pointer;
    transition: all 150ms ease-in-out;
    border: 2px solid var(--foreground-color);

    @media (prefers-color-scheme: dark) {
      border: 2px solid var(--background-color);
    }
```
That kept the black borders on light-mode, but then ALSO kept black borders on dark-mode.

File: src/styles/components/_component-card.scss [hunk 12]
```diff
@@ -48,9 +48,7 @@
       z-index: 1;
       font-size: 1.125rem; // 18px/16
       line-height: 4.6667rem; // 75px
-      letter-spacing: 0.02em;
-      border-top: 0.25rem solid var(--foreground-color);
-      background-color: var(--background-color);
+      background-color: var(--foreground-color);
       position: relative;
       text-align: center;
       display: block;
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:54]: I think we still need these properties that got deleted. I also had to add a quick dark mode query to make sure the headers backgrounds weren't black! 😆 Otherwise, on dark mode, the header background was white ✅, but the light-mode header background was black and you couldn't read the text! 😩

This is what I did starting at line 51:
```
    background-color: var(--background-color);
      position: relative;
      text-align: center;
      display: block;
      letter-spacing: 0.02em;
      border-top: 0.25rem solid var(--foreground-color);

      @media (prefers-color-scheme: dark) {
        border-top: 0.25rem solid var(--background-color);
        background-color: var(--foreground-color);
      }
```

File: src/styles/components/_component-card.scss [hunk 13]
```diff
@@ -77,6 +75,11 @@
         top: 0;
         left: -1rem; // 16px/16
       }
+
+      letter-spacing: 2%;
+      text-align: center;
+      display: block;
+      border-top: 0.25rem solid var(--foreground-color);
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:82]: These can all get deleted because they're getting applied already to the header in about lines 53-ish.

File: src/styles/components/_component-card.scss (hunk 12 above)
Comment (marissahuysentruyt) [reply src/styles/components/_component-card.scss:54]: Oh yeah! This is exactly what my other comment was about! We want the letter-spacing and border, etc to all be here automatically at small screens, so we don't need them in the media query later in the file. They're already here!

File: src/styles/components/_component-grid.scss [hunk 14]
```diff
@@ -30,8 +30,11 @@ $breakpoint-tablet: 820px;
     width: 100%;
     margin-bottom: 1.25rem;
     background-color: var(--light-background-color);
+    margin: 0;
+    background-color: var(--light-background-color);
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-grid.scss:34]: These are repeated properties (we probably missed them during that rebase!), so I would delete them. We want the margin-bottom at line 31.

File: src/styles/components/_footer.scss [hunk 15]
```diff
@@ -64,7 +65,13 @@
...
     &__item > a:hover,
     &__item > a:focus-within {
-      background-color: var(--accent-color);
+      background-color: var(--accent-color-icons);
+    }
+  }
+
+  @media screen and (prefers-color-scheme: dark) {
+    img {
+      filter: invert(100);
```
Comment (marissahuysentruyt) [inline src/styles/components/_footer.scss:74]: Does this need a percent sign for 100%? Or should we use a number between 0 and 1 like the filters a little further down in the file? Like lines 110 or 117? I don't really have a preference either way, but we should do it all the same.

File: src/styles/tools/_mixins.scss [hunk 16]
```diff
@@ -4,6 +4,10 @@ $breakpoint-tablet: 820px;
   background: url("../../../public/images/background_image.png");
   background-size: 100%;
 
+  @media screen and (prefers-color-scheme: dark) {
+    background: url("../../../public/background_image_dark.png");
+  }
+
```
Comment (marissahuysentruyt) [inline src/styles/tools/_mixins.scss:10]: Right now, that new `background_image_dark` is outside of the images folder. We should probably move it, which means we'll have to update this path. 

Also- are you getting this funkiness at super giant screen sizes? Can you see the hole in the tiling? 
Screen Shot 2022-11-08 at 4 30 17 PM

Comment (marissahuysentruyt) [reply src/styles/tools/_mixins.scss:10]: Also, I exported that background-image from Figma once more, as an SVG, and that seemed to help the "busy-ness" that the current image has. It seems to match up better with the light-mode background now. Not sure what difference that made, besides it being an SVG extension. 

```
@media (prefers-color-scheme: dark) {
    background: url("../../../public/images/background_image.svg");
  }
```
Screen Shot 2022-11-08 at 4 41 55 PM
Screen Shot 2022-11-08 at 4 42 05 PM

**Review by marissahuysentruyt — CHANGES_REQUESTED** [review]
@tisidro This is looking SO GOOD after our rebase!! I think the changes I requested probably just snuck through. I tried to suggest how I would fix them too, so hopefully that's helpful!

File: src/styles/components/_footer.scss (hunk 15 above)
Comment (tisidro) [reply src/styles/components/_footer.scss:74]: I think w/ invert you need to use a value 0-100 but I will double check to be sure!

File: src/styles/components/_component-card.scss (hunk 11 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:19]: Thank you so much for figuring this one out!! :) :)

File: src/styles/tools/_mixins.scss (hunk 16 above)
Comment (tisidro) [reply src/styles/tools/_mixins.scss:10]: OK, I am not sure why it wasn't PNG to begin with but it's much better and no weird gap so I'll make sure to download as PNG instead. Thanks for pointing out this issue and a fix! 🥇

File: src/components/RelatedComponents.jsx [hunk 17]
```diff
@@ -6,7 +6,7 @@ export default function RelatedComponents({ components }) {
       <h2 className="rel-component__heading">
         <img
           className="rel-component__heading--arrow"
-          src="/shapes/cmt_inventory_arrow.svg"
+          src="/images/cmt_inventory_arrow.svg"
```
Comment (dustin-jw) [inline src/components/RelatedComponents.jsx:9]: This change is out of scope, and the path is already up-to-date on `main`. Probably the result of rebasing.

File: src/pages/index.page.jsx [hunk 18]
```diff
@@ -17,13 +17,16 @@ export default function Home() {
...
               <img
-                className="home-top__top-image"
                 src="/images/home-header-image_light.svg"
                 alt="Illustration of dog surrounded by accessible components"
               />
-            </div>
+            </picture>
```
Comment (dustin-jw) [inline src/pages/index.page.jsx:29]: The layout of this section has changed, and I wrapping this `picture` in a `div` will fix that. I would also put the `className` back on the `img` tag instead of the `picture` element.

The `picture` element lets us do cool stuff like loading alternate images based on media queries, but the `img` is effectively what is rendered.

File: src/styles/components/_home.scss [hunk 19]
```diff
@@ -43,6 +42,7 @@ $breakpoint-tablet: 820px;
   &-top__top-image {
     width: 100%;
     margin: 3.75rem 0; // top&bottom: 60px/16; left&right: 0
+    background-color: var(--background-color);
```
Comment (dustin-jw) [inline src/styles/components/_home.scss:45]: I don't see a reason to put this here vs. where it was on line 9 before. I'd move it back so it applies to the whole section.

File: src/styles/components/_component-card.scss [hunk 20]
```diff
@@ -6,14 +6,14 @@
 
   .component-card {
     margin: 0;
-    border: 2px solid var(--foreground-color);
+    border: 2px solid var(--background-color);
```
Comment (dustin-jw) [inline src/styles/components/_component-card.scss:9]: I think we want `var(--foreground-color)` here, then `var(--background-color)` in the media query down below for dark mode.

File: src/styles/components/_component-card.scss [hunk 21]
```diff
@@ -48,9 +48,7 @@
       z-index: 1;
       font-size: 1.125rem; // 18px/16
       line-height: 4.6667rem; // 75px
-      letter-spacing: 0.02em;
-      border-top: 0.25rem solid var(--foreground-color);
-      background-color: var(--background-color);
+      background-color: var(--foreground-color);
```
Comment (dustin-jw) [inline src/styles/components/_component-card.scss:51]: I think we want `var(--background-color)` here, then `var(--foreground-color)` down below in the media query for dark mode. We want to avoid looking like this in light mode:

Screenshot 2022-11-09 at 3 47 30 PM

**Review by dustin-jw — CHANGES_REQUESTED** [review]
In addition to @marissahuysentruyt's comments, I have a few suggestions that should fix up some of the bigger problems.

File: src/styles/components/_component-card.scss (hunk 10 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:16]: done! :)

File: src/styles/components/_component-card.scss (hunk 12 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:54]: Yup, it looks good now--thank you! :)

image

File: src/styles/components/_component-card.scss (hunk 13 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:82]: Actually, I didn't see the letter-spacing:2% anywhere else but when I removed it I saw no difference whatsoever. So I guess it's better to not have it!

File: src/styles/components/_component-grid.scss (hunk 14 above)
Comment (tisidro) [reply src/styles/components/_component-grid.scss:34]: Deleted these!

File: src/styles/components/_footer.scss (hunk 15 above)
Comment (tisidro) [reply src/styles/components/_footer.scss:74]: I changed it to 1 per the [MDN guidelines](https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/invert). It goes from 0-1 and values in between are % (like 25 would be written 25%). Thanks for having me check that!

image

File: src/components/RelatedComponents.jsx (hunk 17 above)
Comment (tisidro) [reply src/components/RelatedComponents.jsx:9]: I have learned this the hard way! Thanks and I will always remember that! 🧠

File: src/styles/components/_home.scss (hunk 19 above)
Comment (tisidro) [reply src/styles/components/_home.scss:45]: Done! :)

File: src/styles/components/_component-card.scss (hunk 20 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:9]: Fixed!

File: src/styles/components/_component-card.scss (hunk 21 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:51]: Fixed this too :)

image

File: src/styles/components/_component-card.scss (hunk 13 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss]: You know, on second thought it appears that 2% came from the rebase so it should probably go back in. I just looked on main, and it has been updated to letter-spacing: 2em. So, I think I will just add it back in on my next commit and when we rebase we will just go with whatever is on main.

File: src/pages/index.page.jsx (hunk 18 above)
Comment (tisidro) [reply src/pages/index.page.jsx:29]: Makes sense, I made these changes. Thanks! :)

File: src/styles/components/_component-card.scss [hunk 22]
```diff
@@ -43,10 +46,16 @@
       z-index: 1;
       font-size: 1.125rem; // 18px/16
       line-height: 4.6667rem; // 75px
-      letter-spacing: 0.02em;
-      border-top: 0.25rem solid map.get($colors, "black");
-      background-color: map.get($colors, "white");
+      background-color: var(--background-color);
       position: relative;
+      text-align: center;
+      display: block;
```
Comment (marissahuysentruyt) [inline src/styles/components/_component-card.scss:52]: We'll have to add the `letter-spacing: 0.02em` back in.

File: src/styles/components/_header.scss [hunk 23]
```diff
@@ -30,4 +34,8 @@ $breakpoint-tablet: 820px;
   @media only screen and (min-width: 100rem) {
     margin: auto;
   }
+
+  @media screen and (prefers-color-scheme: dark) {
+    filter: invert(100);
```
Comment (marissahuysentruyt) [inline src/styles/components/_header.scss:39]: This dark-mode query should be deleted. Because of the cascade, this query at line 25 is inverting the Sparkbox logo, making it white. ✅ But then with the duplicated dark-mode query, inverting it again, it's turning it back to black. 😩

File: src/styles/components/_component-card.scss (hunk 22 above)
Comment (tisidro) [reply src/styles/components/_component-card.scss:52]: It was 2%--should I add that back in or just update to the .02em?

File: src/styles/components/_header.scss (hunk 23 above)
Comment (tisidro) [reply src/styles/components/_header.scss:39]: I am going to check this out...but I think we need to just delete the Sparkbox logo altogether based on the design meeting we had. I'll discuss this w/ Dustin at the  pairing we have in a bit.

Comment (tisidro) [reply src/styles/components/_header.scss]: I fixed the logo--I think deleting it would be out of scope here! I also did the feat-dark-mode rebase.  Please re-review when you get a chance! :)

image

File: src/styles/components/_component-card.scss (hunk 22 above)
Comment (marissahuysentruyt) [reply src/styles/components/_component-card.scss:53]: Oh sorry, I put this in a reply above!!

_So the 2% did come from Figma correct. However, Yosevu reminded me that percents done work for letter-spacing, so I grabbed the letter-spacing code from Figma, which was the 0.02em (not 2em)_

**Review by marissahuysentruyt — APPROVED** [review]
This looks great! I would have a second set of eyes check it out one last time, but I think we got all the changes in!!! I just did the rebase on `feat--dark-mode`, so let me know if you need anything while you rebase this branch! I'm open tomorrow morning 

dark-mode

**Review by dustin-jw — APPROVED** [review]
By George I think we've done it! Thank you for sticking it out on this one! 🏆

**Review by dustin-jw — APPROVED** [review]
